bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -1,320 +1,167 @@
                         }
-                    });
-                    content.find('div.ann-title').append(announcementInput);
-                }
-            });
-
-            // Announcement is selected
-            content.find('input[name=announcement]').on('change', function () {
-                const selectedId = parseInt(this.value);
-                const announcement = me.announcements.find(item => item.id === selectedId);
-                // check if announcement is already checked, if is, add/remove accordingly
-                if (!this.checked) {
-                    me.selectedAnnouncements = me.selectedAnnouncements.filter(function (ann) {
-                        return ann.id !== announcement.id;
-                    });
-                } else {
-                    me.selectedAnnouncements.push(announcement);
-                }
-                me.getPlugin().updateAnnouncements(me.selectedAnnouncements);
-                me.updateSelectedInput();
-            });
-
-            popup.show(title, content, [closeButton]);
-            me.announcementsPopup = popup;
-        },
-        // Shows user the currently selected announcement titles next to the tool (informative input/non-functional)
-        updateSelectedInput: function () {
-            jQuery('div.basic_publisher').find('input[name=publisher-announcements]').val(this.selectedAnnouncements.map(i => Oskari.getLocalized(i.locale, this.lang)?.title).toString());
-        },
-        /**
-         * @private @method _getToolPluginAnnouncementsConf
-         * @return {Object / null} config or null if not found
-         */
-        _getToolPluginAnnouncementsConf: function () {
-            var me = this;
-            var isConfig = !!((me.data && me.data.configuration));
-            var isPlugin = !!((isConfig && me.data.configuration.announcements &&
-            me.data.configuration.announcements.conf && me.data.configuration.announcements.conf.plugin));
-            var toolPlugin = null;
-            if (isPlugin) {
-                toolPlugin = me.data.configuration.announcements.conf.plugin;
+                    }
+                });
             }
-            return toolPlugin;
-        },
 
-        _getAnnouncementsSelection: function () {
-            const me = this;
-            const announcementsSelection = {};
-            var pluginValues = me.getPlugin().getSelectedAnnouncements();
-            if (pluginValues.announcements) {
-                announcementsSelection.announcements = pluginValues.announcements;
-            }
-            return announcementsSelection;
-        },
+            this.handler.init({
+                noUI: this.noUI,
+                announcements,
+                selectedAnnouncements
+            });
+        });
+    }
+
+    stop () {
+        this.handler.closePopup();
+        const plugin = this.getPlugin();
+        if (plugin) {
+            plugin.teardownUI();
```

Comment:
Probably need to replace this with something like this: https://github.com/oskariorg/oskari-frontend/blob/master/bundles/framework/publisher2/tools/AbstractPublisherTool.js#L97-L113
or maybe calling `super.stop()` could work?